CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -642,6 +592,65 @@ function Humanoid:hasLeavingAction()
   return false
 end
 
+--! Handle an empty action queue in some way instead of crashing.
+function Humanoid:handleEmptyActionQueue()
+  -- if this is a patient that is going home, an empty
+  -- action queue is not a problem
+  if class.is(self, Patient) and self.going_home then
+    return
+  end
+
+  -- First find out if this humanoid is in a room.
+  local room = self:getRoom()
+  if room then
+    room:makeHumanoidLeave(self)
+  end
+
+  -- Give the humanoid an action to avoid crashing.
+  if class.is(self, Staff) then
+    self:queueAction(MeanderAction())
+  elseif class.is(self,GrimReaper) then
+    self:queueAction(IdleAction())
+  else
+    self:queueAction(SeekReceptionAction())
+  end
+
+  -- Open the dialog of the humanoid as feedback to the user.
+  local ui = self.world.ui
+  if class.is(self, Patient) then
+    ui:addWindow(UIPatient(ui, self))
+  elseif class.is(self, Staff) then
+    ui:addWindow(UIStaff(ui, self))
+  end
+
+  -- Tell the player what just happened.
+  self.world:gameLog("")
+  self.world:gameLog("Empty action queue!")
+  self.world:gameLog("Last action: " .. self.previous_action.name)
+  self.world:gameLog(debug.traceback())
+
+  ui:addWindow(UIConfirmDialog(ui, true,
+    "Sorry, a humanoid just had an empty action queue,"..
+    " which means that he or she didn't know what to do next."..
+    " Please consult the command window for more detailed information. "..
+    "A dialog with "..
+    "the offending humanoid has been opened. "..
+    "Would you like him/her to leave the hospital?",
+    --[[persistable:humanoid_leave_hospital]] function()
```

Comment:
Would this be game-breaking if someone saved the game on this error (before this PR) and reloaded with the new code? -- I don't think it should.

Reply:
I agree with you. Persistence doesn't care about scope afaik.

Reply:
It's fne